packages/typespec-azure-core/lib/models.tsp

```diff
@@ -349,6 +349,17 @@ scalar eTag extends string;
  */
 scalar azureLocation extends string;
 
+/**
+ * Represents an Azure Resource Type.
+ *
+ * @example
+ *
+ * ```
+ * Microsoft.Network/virtualNetworks/subnets
+ * ```
+ */
+scalar resourceType extends string;
```

Comment:
Sorry, I was OOF the last 2 weeks, has there been agreement we needed this and this was the name we wanted?

Reply:
Oh - I do not know. Do we need to bring up this topic in one of our typespec sync up meeting or your internal typespec sync up meeting?
In those meetings that I have participated in, this has not been a topic there.
Or maybe we could put this as an agenda in our next typespec sync up meeting on your wednesday night.

Reply:
I think we probably need to differentiate this as we did with armResourceIdentifier - armResourceType.  

There is definitely use for this in ARM, and since the type is needed to properly specify the allowedResourceType in armResourceIdentifier, it seems appropriate to put this here.

Reply:
Yeah I agree the name should be `armResourceType` which makes more sense.
The second part is that you suggest we add a generic argument for allowed resource type? like this?
```
scalar armResourceType<AllowedResourceTypes extends ArmResourceIdentifierAllowedResource[] = never> extends string {}
```
I think this looks weird because this type is modeling the resource type of myself, if there is a limited list of resource types that this field is supporting, should we define a extensible enum based on resource type like this?
```
union SupportedResourceType extends armResourceType {
	"Microsoft.Compute/virtualMachines"
}
```
Well but I could see values for this because we could update the definition of `Resource` and `TrackedResource` to let them accept another argument of resource type like this:
```
model TrackedResource<Properties extends Model, ResourceType extends ArmResourceIdentifierAllowedResource> {
id: armResourceIdentifier;
type: armResourceType<ResourceType>;

properties: Properties;
}
```
so that we finally have explicitly defined resource type on resources, currently arm resources could have resource types but they are implicitly - we can infer them from the uri paths we built for resource related operations.

Reply:
yeah I think using a union here instead to be more precise makes more sense instead of the `armREsourceIdentifier` which does that for a subset of the value.

and `+1` on the name being `armResourceType`